File: LogAndroid/TrustarApp/src/main/kotlin/jp/co/toukei/log/trustar/db/chat/entity/ChatRoomExt.kt
package jp.co.toukei.log.trustar.db.chat.entity

import androidx.room.Embedded
import androidx.room.Ignore
import jp.co.toukei.log.lib.millisAfterOffset
import jp.co.toukei.log.lib.unixTimeDaysBetween
import jp.co.toukei.log.trustar.Config
import jp.co.toukei.log.trustar.imageURI
import third.Clock

class ChatRoomExt(
    @Embedded @JvmField val room: ChatRoom,
) {

    @Ignore
    @JvmField
    val imageUri = room.image?.imageURI()

    @Ignore
    @JvmField
    val title = "${room.name}(${room.userCount})"

    @Ignore
    @JvmField
    val lastUpdateHHmm = Clock(Config.timeZone.millisAfterOffset(room.lastUpdated)).hhmm

    @Ignore
    @JvmField
    val lastUpdateMMdd = Config.mmddFormatter.format(room.lastUpdated)

    fun sameDayToLastUpdated(time: Long): Boolean {
        return Config.timeZone.unixTimeDaysBetween(room.lastUpdated, time) == 0
    }
}

File: LogiOS/TrustarApp/Database/chat/Entity/ChatRoomExt.swift
//
//  ChatRoomExt.swift
//  TrustarApp
//
//  Created by DionSoftware on 2022/11/24.
//  Copyright © 2022 DionSoftware. All rights reserved.
//

import Foundation

class ChatRoomExt {
    var room: ChatRoom
    init(room: ChatRoom) {
        self.room = room
    }
    
    func imageUri() -> ImageURI? {
        if let img = room.image {
            if let url = URL(string: img) {
                return ImageURI(uri: url, handle: true)
            }
        }
        
        return nil
    }

    func title() -> String {
        return "\(room.name ?? "")(\(room.userCount))"
    }

    func lastUpdateHHmm() -> String {
        return Config.Shared.dateFormatterHHmm.format(date: Date(timeIntervalSince1970: TimeInterval(room.lastUpdated / 1000)))
    }

    func lastUpdateMMdd() -> String {
        return Config.Shared.dateFormatterMMdd.format(date: Date(timeIntervalSince1970: TimeInterval(room.lastUpdated / 1000)))
    }

    func sameDayToLastUpdated(time: Int64) -> Bool {
        return daysBetween(from: room.lastUpdated, to: time) == 0
    }
}
